sockeye/utils.py

```diff
@@ -710,3 +711,16 @@ def create_operator(self, ctx, shapes, dtypes):
         return PrintValue(self.print_name,
                           print_grad=self.print_grad,
                           use_logger=self.use_logger)
+
+
+def grouper(iterable: Iterable, size: int) -> Iterable:
+    """
+    Collect data into fixed-length chunks or blocks
+    without either dicarding underfilled chunks or padding them
```

Comment:
missing period.
missing param doc string.

Also, why the 'either'? Why not: "without discarding underfilled chunks or padding them".
Maybe explicitly say that the last chunk might be smaller than size.